manifests/logical_volume.pp

```diff
@@ -2,73 +2,84 @@
-  Boolean $createfs                 = true,
-  $extents                          = undef,
-  $stripes                          = undef,
-  $stripesize                       = undef,
-  $readahead                        = undef,
-  $range                            = undef,
-  $size_is_minsize                  = undef,
-  $type                             = undef,
-  $thinpool                         = false,
-  $poolmetadatasize                 = undef,
-  $mirror                           = undef,
-  $mirrorlog                        = undef,
-  $no_sync                          = undef,
-  $region_size                      = undef,
-  $alloc                            = undef,
+  $size                                               = undef,
+  $initial_size                                       = undef,
+  Enum['absent', 'present'] $ensure                   = present,
+  $options                                            = 'defaults',
+  $pass                                               = '2',
+  $dump                                               = '0',
+  $fs_type                                            = 'ext4',
+  $mkfs_options                                       = undef,
+  Variant[Enum['/'], Stdlib::Absolutepath] $mountpath = "/${name}",
+  Boolean $mountpath_require                          = false,
+  Boolean $mounted                                    = true,
+  Boolean $createfs                                   = true,
+  $extents                                            = undef,
+  $stripes                                            = undef,
+  $stripesize                                         = undef,
+  $readahead                                          = undef,
+  $range                                              = undef,
+  $size_is_minsize                                    = undef,
+  $type                                               = undef,
+  $thinpool                                           = false,
+  $poolmetadatasize                                   = undef,
+  $mirror                                             = undef,
+  $mirrorlog                                          = undef,
+  $no_sync                                            = undef,
+  $region_size                                        = undef,
+  $alloc                                              = undef,
+  $owner                                              = undef,
+  $group                                              = undef,
+  $mode                                               = undef,
 ) {
 
   $lvm_device_path = "/dev/${volume_group}/${name}"
 
   if $mountpath_require and $fs_type != 'swap' {
+
+    file { $mountpath:
+      ensure => directory,
+      group  => $group,
+      owner  => $owner,
+      mode   => $mode,
+    }
+
     Mount {
       require => File[$mountpath],
     }
   }
 
   if $fs_type == 'swap' {
-    $mount_title     = $lvm_device_path
+    $mount_title = $lvm_device_path
```

Comment:
Yeah, the unfortunate things IntelliJ ctrl+alt+lowercase L does

I can change it back if needed